src/coreclr/nativeaot/Runtime/EHHelpers.cpp

```diff
@@ -336,22 +336,22 @@ static bool InWriteBarrierHelper(uintptr_t faultingIP)
         ASSERT(*(uint8_t*)writeBarrierAVLocations[i] != 0xE9); // jmp XXXXXXXX
 #endif
 
-        if (PCODEToPINSTR(writeBarrierAVLocations[i]) == faultingIP)
+        if (writeBarrierAVLocations[i] == faultingIP)
             return true;
     }
 #endif // USE_PORTABLE_HELPERS
 
     return false;
 }
 
-EXTERN_C void* RhpInitialInterfaceDispatch;
-EXTERN_C void* RhpInterfaceDispatchAVLocation1;
-EXTERN_C void* RhpInterfaceDispatchAVLocation2;
-EXTERN_C void* RhpInterfaceDispatchAVLocation4;
-EXTERN_C void* RhpInterfaceDispatchAVLocation8;
-EXTERN_C void* RhpInterfaceDispatchAVLocation16;
-EXTERN_C void* RhpInterfaceDispatchAVLocation32;
-EXTERN_C void* RhpInterfaceDispatchAVLocation64;
+EXTERN_C CODE_LOCATION RhpInitialInterfaceDispatch;
```

Comment:
Side-node: This is broken on ARM. Firstly, it is function entrypoint and not AV location. Secondly, we don't do the access in the assembly that would cause the AV.

I'll probably submit the fix separately since I cannot test it on device now.